src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -257,8 +266,35 @@ private void registerFabUploadListeners() {
         getFabUpload().setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                UploadFilesActivity.startUploadActivityForResult(getActivity(), ((FileActivity) getActivity())
-                        .getAccount(), FileDisplayActivity.REQUEST_CODE__SELECT_FILES_FROM_FILE_SYSTEM);
+                Log_OC.e(TAG,"Clicked" + getContext().toString());
+                View uploadBottomSheet = getLayoutInflater().inflate(R.layout.upload_bottom_sheet_fragment,null);
+                final BottomSheetDialog dialog = new BottomSheetDialog(getContext());
+                dialog.setContentView(uploadBottomSheet);
+                uploadFilesLinearLayout = (LinearLayout) uploadBottomSheet.findViewById(R.id.files_linear_layout);
+                uploadFromCameraLinearLayout = (LinearLayout) uploadBottomSheet.findViewById(R.id.upload_from_camera_linear_layout);
+                uploadToTextView = (TextView) uploadBottomSheet.findViewById(R.id.upload_to_text_view);
+                uploadFilesLinearLayout.setOnTouchListener(new View.OnTouchListener() {
+                    @Override
+                    public boolean onTouch(View v, MotionEvent event) {
+                        UploadFilesActivity.startUploadActivityForResult(getActivity(), ((FileActivity) getActivity())
+                                .getAccount(), FileDisplayActivity.REQUEST_CODE__SELECT_FILES_FROM_FILE_SYSTEM);
+                        dialog.hide();
+                        return false;
+                    }
+                });
+                uploadFromCameraLinearLayout.setOnTouchListener(new View.OnTouchListener() {
+                    @Override
+                    public boolean onTouch(View v, MotionEvent event) {
+                        UploadFilesActivity.startUploadActivityForResult(getActivity(),((FileActivity) getActivity()).getAccount(),
+                                FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA);
+                        dialog.hide();
+                        return false;
+                    }
+                });
+                uploadToTextView.setText(String.format(getResources().getString(R.string.upload_to),getResources().getString(R.string.app_name)));
+                uploadBottomSheetBehavior = BottomSheetBehavior.from((View) uploadBottomSheet.getParent());
+                uploadBottomSheetBehavior.setPeekHeight(-1);
```

Comment:
I've managed to fix https://github.com/owncloud/android/pull/2097#issuecomment-363387082 and  https://github.com/owncloud/android/pull/2097#issuecomment-363398512 to give you a hand, follow the next steps:

1. Delete line `uploadBottomSheetBehavior.setPeekHeight(-1);`
2. Include the next code:

```
dialog.setOnShowListener(new DialogInterface.OnShowListener() {
    @Override
    public void onShow(DialogInterface dialog) {
        uploadBottomSheetBehavior.setPeekHeight(uploadBottomSheet.getMeasuredHeight());
    }
});
```
With this code, we wait till the dialog is shown, measure its height and use it to set the height of the bottom sheet. I've tested it with different devices (portrait & landscape) and seems to be working properly.

Reply:
@davigonz Thanks for the help. I trying to find the value of the height of bottom sheet using ViewTreeObserver class but it did not work out. I will implement the changes.